benchmarks/test_benchmark_coo.py

```diff
@@ -77,65 +77,47 @@ def bench():
         f(x, y)
 
 
-@pytest.fixture(params=[100, 500, 1000], ids=side_ids)
-def indexing_args(request, seed):
-    side = request.param
-    if side**3 >= 2**26:
+@pytest.fixture(params=itertools.product([100, 500, 1000], [1, 2, 3]), ids=get_test_id)
+def indexing_args(request, seed, max_size):
+    side, rank = request.param
+    if side**rank >= max_size:
         pytest.skip()
     rng = np.random.default_rng(seed=seed)
+    shape = (side,) * rank
 
-    return sparse.random((side, side, side), density=DENSITY, random_state=rng)
+    return sparse.random(shape, density=DENSITY, random_state=rng)
 
 
 def test_index_scalar(benchmark, indexing_args):
     x = indexing_args
     side = x.shape[0]
+    rank = x.ndim
 
-    x[side // 2, side // 2, side // 2]  # Numba compilation
+    x[(side // 2,) * rank]  # Numba compilation
 
     @benchmark
     def bench():
-        x[side // 2, side // 2, side // 2]
+        x[(side // 2,) * rank]
 
 
 def test_index_slice(benchmark, indexing_args):
     x = indexing_args
     side = x.shape[0]
+    rank = x.ndim
 
-    x[: side // 2]  # Numba compilation
-
-    @benchmark
-    def bench():
-        x[: side // 2]
-
-
-def test_index_slice2(benchmark, indexing_args):
-    x = indexing_args
-    side = x.shape[0]
-
-    x[: side // 2, : side // 2]  # Numba compilation
-
-    @benchmark
-    def bench():
-        x[: side // 2, : side // 2]
-
-
-def test_index_slice3(benchmark, indexing_args):
-    x = indexing_args
-    side = x.shape[0]
-
-    x[: side // 2, : side // 2, : side // 2]  # Numba compilation
+    x[(slice(side // 2),) * rank]  # Numba compilation
 
     @benchmark
     def bench():
-        x[: side // 2, : side // 2, : side // 2]
+        x[(slice(side // 2),) * rank]
 
 
 def test_index_fancy(benchmark, indexing_args, seed):
     x = indexing_args
     side = x.shape[0]
+    rank = x.ndim
     rng = np.random.default_rng(seed=seed)
-    index = rng.integers(0, side, side // 2)
+    index = rng.integers(0, side, size=(side // 2,) * rank)
```

Comment:
This issue is here -- we need to make this array 1D as `sparse` doesn't yet support N-dimensional array indices. We can remove the ` * rank` here.

Reply:
Done. Thanks.